### PR TITLE
Introduce new logger configuration

### LINKED ISSUE #481: Logger Stream
`Hanami::Logger` supports only stdout as output stream. This has [several advantages](http://12factor.net/logs) such as leaving the freedom to ops team to redirect the stream like they want, or to not accumulate endless (and useless) log files during development.

We should keep stdout as default and preferred stream to log informations.

However, we want to make this more flexible and let developers to **use their own logger**, or **write to file**. In order to make this possible, we want to introduce a new setting:

```ruby
# apps/web/config/application.rb
configure do
  logger.stream 'path/to/file'
end
```

The argument can be:

  * `String`, `Pathname` with relative or absolute path to file
  * A stream like instances of `IO`, `StringIO` like `$stdout` or `STDOUT`

**It MUST default to `STDOUT`.**

### Generating Apps

Projects with default generated app (`Web`) and new generated app (`hanami generate app`), MUST include that setting in the **main** `configure` block **as a commented line of code**.

### Upgrading Existing Projects

Projects that will upgrade to this new version will still use stdout, as it will remain the default output.

### PR DESCRIPTION
**Pull request by davydovanton**
### TODO list

- [X] Create new `Config::Logger` class
- [X] Create `stream` and `build` methods
- [x] Returns `Config::Logger` instance in `logger` config option and fix `_configure_logger!` logic
- [x] Update configuration file
- [ ] Add informations about logger to guides

/cc @hanami/core-team 

---

Closes #481

### CODE REVIEW

File: lib/hanami/config/logger.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      end
+
+      # Returns new Hanami::Logger instance with all options
+      #
+      # @return [Hanami::Logger] the new logger instance
+      #
+      # @since x.x.x
+      # @api private
+      def build
+        ::Hanami::Logger.new(log_device: @log_device)
```
Comment (jodosha) [inline lib/hanami/config/logger.rb]: @davydovanton Now it's `device:`, not `log_device:` :wink:

**Comment by davydovanton** [comment]
@jodosha could you check this?

**Comment by joneslee85** [comment]
@davydovanton looks :+1:

**Comment by jodosha** [comment]
@davydovanton It's really good. Thank you. Merging it! :+1:

**Pull request merged**